app/controllers/api/vms_controller.rb

```diff
@@ -53,6 +54,98 @@ def edit_resource(type, id, data)
     def delete_resource(type, id, _data = nil)
       enqueue_ems_action(type, id, "Deleting", :method_name => "destroy")
     end
+ 
+    def vm_reconfigure_resource(type, id, data)
+    
+      new_row = {
+        :id        => id.to_s,
+        :long_id   => id.to_s,
+        :cells     => []
+      }
+      
+      unless id
+          data_spec = data.collect { |key, val| "#{key}=#{val}" }.join(", ")
+          raise NotFoundError, "Invalid #{type} resource specified - #{data_spec}"
+        end
+        resource = resource_search(id, type)
+        new_row = vm_data_collection(new_row,resource)
+        
+        opts = {
+          :name                  => type.to_s,
+          :is_subcollection      => false,
+          :expand_resources      => true,
+          :expand_actions        => true,
+          :expand_custom_actions => true
+        }
+        resource_to_jbuilder(type, type, resource, opts).attributes!
+    end
+
+    def vm_data_collection(new_row,resource)
+      new_row[:tree_id] = TreeBuilder.build_node_id(resource) if resource
+      new_row[:cells] << {:is_checkbox => true}
+      new_row[:cells] << {:name => resource.name}
+      new_row[:cells] << {:hostname => resource&.host&.name}
+      new_row[:cells] << {:num_cpu => resource.num_cpu}
+      new_row[:cells] << {:cpu_cores_per_socket => resource.cpu_cores_per_socket}
+      new_row[:cells] << {:mem_cpu => "#{(resource.mem_cpu.to_i / 1024.0).to_i}GB"}
```

Comment:
Some of these check if resource exists.  Suggest we do `resource&.` for each in case resource is nil.  Or, should we return if `resource.nil?` or raise an exception?